### PR TITLE
doc: Adds Changelog for v1.15.2

### PR DESCRIPTION
**Pull request by maastha**
## Description

This PR adds Changelog for v1.15.2 release.

Link to any related issue(s): https://jira.mongodb.org/browse/CLOUDP-237981 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [x] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,123 @@
 # Changelog
 
+## [v1.15.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.15.2) (2024-03-15)
+
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.15.1...v1.15.2-pre)
+
+**Enhancements**
+
+- feat: Adds support for Sample stream type to `mongodbatlas_stream_connection` [\#2026](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2026) ([maastha](https://github.com/maastha))
+- feat: Add support for using DEV/QA for mongodbgov [\#2009](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2009) ([maastha](https://github.com/maastha))
```
Comment (lantoli) [inline CHANGELOG.md]:
```suggestion
- feat: Adds support for using DEV/QA for mongodbgov [\#2009](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2009) ([maastha](https://github.com/maastha))
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,123 @@
...
+- chore: Unifies pass of org variables in tests [\#1997](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1997) ([lantoli](https://github.com/lantoli))
+- chore: Stops using Network org in tests [\#1996](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1996) ([lantoli](https://github.com/lantoli))
+- chore: Changes skip project logic for cleanup action [\#1995](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1995) ([lantoli](https://github.com/lantoli))
+- chore: Bump dorny/paths-filter from 3.0.1 to 3.0.2 [\#1994](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1994) ([dependabot[bot]](https://github.com/apps/dependabot))
+- chore: Bump tj-actions/verify-changed-files from cb704d8c717959cf92ec9db9131372bc20356aa2 to 79a8cef5d9ef3ab541ee07ef179bd6c3c2d42ecc [\#1993](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1993) ([dependabot[bot]](https://github.com/apps/dependabot))
+- chore: Bump peter-evans/create-pull-request from 6.0.0 to 6.0.1 [\#1992](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1992) ([dependabot[bot]](https://github.com/apps/dependabot))
```
Comment (lantoli) [inline CHANGELOG.md]: suggest to move all bumps together at the end of the section

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,123 @@
 # Changelog
 
+## [v1.15.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.15.2) (2024-03-15)
+
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.15.1...v1.15.2-pre)
```
Comment (lantoli) [inline CHANGELOG.md]:
```suggestion
[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.15.1...v1.15.2)
```

**Review by lantoli — APPROVED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,123 @@
 # Changelog
 
+## [v1.15.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.15.2) (2024-03-15)
```
Comment (andreaangiolillo) [inline CHANGELOG.md:3]: Should we also update the link to the example in the doc main page to point to the new tag?

**Review by Zuhairahmed — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (maastha) [reply CHANGELOG.md:3]: not sure if we reached a consensus on that, @AgustinBettati could you confirm?

File: CHANGELOG.md (hunk 2 above)
Comment (maastha) [reply CHANGELOG.md]: done

**Review by oarbusi — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (maastha) [reply CHANGELOG.md:3]: discussed offline, we'll update the link and outline the procedure during release with a separate ticket for future releases

**Review by jwilliams-mongo — COMMENTED** [review]
LGTM

**Pull request merged**